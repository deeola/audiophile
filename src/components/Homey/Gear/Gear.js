import React from "react";
import {
  GearContainer,
  Left,
  Right,
  Text,
  Title,
  Paragraph,
} from "./Gearstyling";

const Gear = () => {
  return (
    <GearContainer data-testid='gear'>
      <Left>
        <Text>
          <Title>
            Bringing you the <span>best</span> audio gear
          </Title>
          <Paragraph>
            Located at the heart of New York City, Audiophile is the premier
            store for high end headphones, earphones, speakers, and audio
            accessories. We have a large showroom and luxury demonstration rooms
            available for you to browse and experience a wide range of our
            products. Stop by our store to meet some of the fantastic people who
            make Audiophile the best place to buy your portable audio equipment.
          </Paragraph>
        </Text>
      </Left>
      <Right />
    </GearContainer>
  );
};

export default Gear;
