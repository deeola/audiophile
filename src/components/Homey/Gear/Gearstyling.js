import styled from "styled-components";
import desktop from "../../../assets/shared/desktop/gearimage.jpg";
import tablet from "../../../assets/shared/tablet/tabgear.jpg";
import mobile from "../../../assets/shared/mobile/mobilegear.jpg";

export const GearContainer = styled.section`
  display: flex;
  justify-content: space-between;
  width: 100%;
  margin: 50px auto;

  @media (max-width: 802px) {
    flex-direction: column-reverse;
  }

  @media (max-width: 490px) {
  }
`;

export const Left = styled.div`
  width: 49%;
  margin: 130px 0px;
  @media (max-width: 802px) {
    width: 100%;
    margin: 0px 0px;
  }

  @media (max-width: 490px) {
  }
`;

//use window width to deternine image source

export const Right = styled.div`
  width: 49%;
  background-image: url(${desktop});
  position: relative;
  background-repeat: no-repeat;
  background-size: cover;
  content: "";
  position: relative;
  @media (max-width: 802px) {
    width: 100%;
    background-image: url(${tablet});
    height: 300px;
    background-position: center;
  }

  @media (max-width: 490px) {
    background-image: url(${mobile});
  }
`;

export const Text = styled.div`
  width: 90%;
  @media (max-width: 802px) {
    width: 100%;
  }
`;

export const Title = styled.h2`
  text-transform: uppercase;
  font-size: 40px;
  width: 80%;
  margin: 30px 0px;
  span {
    color: #d87d4a;
  }
  @media (max-width: 802px) {
    text-align: center;
    margin: 30px auto;
  }

  @media (max-width: 490px) {
    width: 100%;
    font-size: 36px;
    margin: 20px 0px;
  }
`;

export const Paragraph = styled.p`
  @media (max-width: 802px) {
    text-align: center;
    width: 90%;
    margin: auto;
    line-height: 27px;
  }

  @media (max-width: 490px) {
    width: 100%;
  }
`;
