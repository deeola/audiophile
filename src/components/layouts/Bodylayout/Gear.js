import React from "react";
import tabgear from "../../../assets/shared/tablet/tabgear.jpg";
import imageearphones from "../../../assets/shared/desktop/gearimage.jpg";
import mobilegear from "../../../assets/shared/mobile/mobilegear.jpg";

function Gear(props) {
  return (
    <div className="Gear">
      <div className="tabgear">
        <img src={tabgear} alt="man with headphone"></img>
      </div>
      <div className="mobilegear">
        <img src={mobilegear} alt="man with headphone"></img>
      </div>
      <div className="GearTextDiv">
        <div className="ggtext">
          <p className="gearmaintitle">
            Bringing you the <span>best</span> audio gear
          </p>
          <p className="gmaintext">
            {" "}
            Located at the heart of New York City, Audiophile is the premier
            store for high end headphones, earphones, speakers, and audio
            accessories. We have a large showroom and luxury demonstration rooms
            available for you to browse and experience a wide range of our
            products. Stop by our store to meet some of the fantastic people who
            make Audiophile the best place to buy your portable audio equipment.
          </p>
        </div>
      </div>
      <div className="gearImageDiv">
        <img src={imageearphones} alt="man with headphone"></img>
      </div>
    </div>
  );
}

export default Gear;
