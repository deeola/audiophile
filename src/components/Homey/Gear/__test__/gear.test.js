import { render, cleanup } from "@testing-library/react";
import reactDom from "react-dom";
import Gear from "../Gear";
import renderer from "react-test-renderer";

it("renders div correctly", () => {
  const div = document.createElement("div");

  reactDom.render(<Gear />, div);
});

it("contains a span element", () => {
  const { getByTestId } = render(<Gear />);
  expect(getByTestId("gear")).toContainHTML("span");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Gear />).toJSON();

  expect(tree).toMatchSnapshot();
});
