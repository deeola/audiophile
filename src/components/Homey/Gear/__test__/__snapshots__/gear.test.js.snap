// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="sc-bczRLJ ipeuwb"
  data-testid="gear"
>
  <div
    className="sc-gsnTZi qkPPL"
  >
    <div
      className="sc-hKMtZM hXoByT"
    >
      <h2
        className="sc-eCYdqJ jCwkCF"
      >
        Bringing you the 
        <span>
          best
        </span>
         audio gear
      </h2>
      <p
        className="sc-jSMfEi"
      >
        Located at the heart of New York City, Audiophile is the premier store for high end headphones, earphones, speakers, and audio accessories. We have a large showroom and luxury demonstration rooms available for you to browse and experience a wide range of our products. Stop by our store to meet some of the fantastic people who make Audiophile the best place to buy your portable audio equipment.
      </p>
    </div>
  </div>
  <div
    className="sc-dkzDqf eaVlON"
  />
</section>
`;
